{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Task-1.-Compiling-Ebola-Data\"><span class=\"toc-item-num\">Task 1.&nbsp;&nbsp;</span>Compiling Ebola Data</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-2.-RNA-Sequences\"><span class=\"toc-item-num\">Task 2.&nbsp;&nbsp;</span>RNA Sequences</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-3.-Class-War-in-Titanic\"><span class=\"toc-item-num\">Task 3.&nbsp;&nbsp;</span>Class War in Titanic</a></div></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "DATA_FOLDER = '../../Tutorials/02 - Intro to Pandas/Data/' # Use the data folder provided in Tutorial 02 - Intro to Pandas."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 1. Compiling Ebola Data\n",
    "\n",
    "The `DATA_FOLDER/ebola` folder contains summarized reports of Ebola cases from three countries (Guinea, Liberia and Sierra Leone) during the recent outbreak of the disease in West Africa. For each country, there are daily reports that contain various information about the outbreak in several cities in each country.\n",
    "\n",
    "Use pandas to import these data files into a single `Dataframe`.\n",
    "Using this `DataFrame`, calculate for *each country*, the *daily average per month* of *new cases* and *deaths*.\n",
    "Make sure you handle all the different expressions for *new cases* and *deaths* that are used in the reports."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Potential problems:\n",
    "- Column name (same for rows) + inconsistent case sensitive\n",
    "Date,Variable,National\n",
    "date,variable,...National\n",
    "Date,Description,Totals\n",
    "\n",
    "- Split rows\n",
    "12/1/2014,\"Total death/s in confirmed, \n",
    " probable, suspected cases\",3161,158,124,10,63,74,5,26,384,549,17,1651,53,8,25,14,\n",
    " \n",
    "- Unnamed/partial columns\n",
    "- Some rows are counts of others\n",
    "New cases of suspects, New cases of probables, New cases of confirmed, Total new cases registered so far"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import glob\n",
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "countries_paths = glob.glob(DATA_FOLDER + 'ebola/*/')\n",
    "countries_names = [os.path.basename(os.path.normpath(x)).replace('_data', '').capitalize() for x in countries]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Guinea', 'Liberia', 'Sl']\n"
     ]
    }
   ],
   "source": [
    "print(countries_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "aliases = {\n",
    "    'variable': 'description',\n",
    "    'national': 'totals'\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "dfs = []\n",
    "for i, path in enumerate(countries_paths):\n",
    "    filenames = glob.glob(path + \"/*.csv\")\n",
    "    \n",
    "    for f in filenames:\n",
    "        df = pd.read_csv(f)\n",
    "        df.columns = [c.lower() for c in df.columns] # Set all columns to lower-case\n",
    "        df.rename(columns=aliases, inplace=True)\n",
    "        df['country'] = countries_names[i]\n",
    "        df['date'] = [parser.parse(x, tzinfos={'dayfirst': True}) for x in df['date']]\n",
    "        df['totals'] = pd.to_numeric(df['totals'], errors='coerce')\n",
    "        df = df.fillna(0)\n",
    "        dfs.append(df[['date', 'country', 'description', 'totals']])\n",
    "        \n",
    "df = pd.concat(dfs)\n",
    "df['description'] = [x.lower() for x in df['description']]\n",
    "#print(df['description'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "from itertools import product\n",
    "\n",
    "column_keywords = {\n",
    "    'new_cases': ['new cases of suspects', 'new cases of probables', 'new cases of confirmed', 'new case/s (suspected)',\n",
    "                 'new case/s (probable)', 'new case/s (confirmed)',  'new_suspected', 'new_probable', 'new_confirmed'],\n",
    "    'new_deaths': ['new deaths registered today', 'new deaths registered', 'newly reported deaths',\n",
    "                  'etc_new_deaths']\n",
    "}\n",
    "\n",
    "desc_map = []\n",
    "for k, v in column_keywords.items():\n",
    "    for alias in v:\n",
    "        desc_map.append((k, alias))\n",
    "desc_map = pd.DataFrame(desc_map, columns=['type', 'description'])\n",
    "df = df.merge(desc_map, on='description', how='inner')\n",
    "df = df.drop('description', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>country</th>\n",
       "      <th>totals</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>5.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2014-08-26</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>18.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2014-08-27</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>12.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2014-08-30</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>15.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014-08-31</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>9.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        date country  totals       type\n",
       "0 2014-08-04  Guinea     5.0  new_cases\n",
       "1 2014-08-26  Guinea    18.0  new_cases\n",
       "2 2014-08-27  Guinea    12.0  new_cases\n",
       "3 2014-08-30  Guinea    15.0  new_cases\n",
       "4 2014-08-31  Guinea     9.0  new_cases"
      ]
     },
     "execution_count": 283,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pivoted = df.pivot_table(index=['date', 'country'], columns='type', values='totals', aggfunc='sum')\n",
    "#pivoted = pivoted['totals'].reset_index()\n",
    "pivoted.columns.name = None\n",
    "df = pivoted.reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>country</th>\n",
       "      <th>type</th>\n",
       "      <th>totals</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2014-06-16</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>new_cases</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2014-06-16</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>new_deaths</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2014-06-17</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>new_cases</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2014-06-17</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>new_deaths</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014-06-22</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>new_cases</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        date  country        type  totals\n",
       "0 2014-06-16  Liberia   new_cases     4.0\n",
       "1 2014-06-16  Liberia  new_deaths     2.0\n",
       "2 2014-06-17  Liberia   new_cases     2.0\n",
       "3 2014-06-17  Liberia  new_deaths     0.0\n",
       "4 2014-06-22  Liberia   new_cases    10.0"
      ]
     },
     "execution_count": 243,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from dateutil import parser\n",
    "df_ = df.groupby([df.date, df.country, df.type]).sum().reset_index()\n",
    "df_.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>country</th>\n",
       "      <th>new_cases</th>\n",
       "      <th>new_deaths</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2014-06-16</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2014-06-17</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2014-06-22</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>10.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2014-06-24</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014-06-25</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>7.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        date  country  new_cases  new_deaths\n",
       "0 2014-06-16  Liberia        4.0         2.0\n",
       "1 2014-06-17  Liberia        2.0         0.0\n",
       "2 2014-06-22  Liberia       10.0         4.0\n",
       "3 2014-06-24  Liberia        6.0         4.0\n",
       "4 2014-06-25  Liberia        7.0         3.0"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.country == 'Liberia'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>country</th>\n",
       "      <th>year</th>\n",
       "      <th>month</th>\n",
       "      <th>new_cases</th>\n",
       "      <th>new_deaths</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>25.800000</td>\n",
       "      <td>3.400000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>2014</td>\n",
       "      <td>9</td>\n",
       "      <td>19.625000</td>\n",
       "      <td>3.562500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>2014</td>\n",
       "      <td>10</td>\n",
       "      <td>34.000000</td>\n",
       "      <td>15.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>6</td>\n",
       "      <td>5.714286</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>7</td>\n",
       "      <td>8.545455</td>\n",
       "      <td>4.272727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>37.222222</td>\n",
       "      <td>23.222222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>9</td>\n",
       "      <td>63.833333</td>\n",
       "      <td>36.041667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>10</td>\n",
       "      <td>45.560000</td>\n",
       "      <td>28.040000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>11</td>\n",
       "      <td>26.466667</td>\n",
       "      <td>13.466667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>12</td>\n",
       "      <td>5178.555556</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Sl</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>25.150000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Sl</td>\n",
       "      <td>2014</td>\n",
       "      <td>9</td>\n",
       "      <td>40.689655</td>\n",
       "      <td>0.275862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Sl</td>\n",
       "      <td>2014</td>\n",
       "      <td>10</td>\n",
       "      <td>70.928571</td>\n",
       "      <td>3.535714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Sl</td>\n",
       "      <td>2014</td>\n",
       "      <td>11</td>\n",
       "      <td>75.238095</td>\n",
       "      <td>0.571429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Sl</td>\n",
       "      <td>2014</td>\n",
       "      <td>12</td>\n",
       "      <td>41.000000</td>\n",
       "      <td>2.200000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    country  year  month    new_cases  new_deaths\n",
       "0    Guinea  2014      8    25.800000    3.400000\n",
       "1    Guinea  2014      9    19.625000    3.562500\n",
       "2    Guinea  2014     10    34.000000   15.000000\n",
       "3   Liberia  2014      6     5.714286    2.000000\n",
       "4   Liberia  2014      7     8.545455    4.272727\n",
       "5   Liberia  2014      8    37.222222   23.222222\n",
       "6   Liberia  2014      9    63.833333   36.041667\n",
       "7   Liberia  2014     10    45.560000   28.040000\n",
       "8   Liberia  2014     11    26.466667   13.466667\n",
       "9   Liberia  2014     12  5178.555556    0.000000\n",
       "10       Sl  2014      8    25.150000    0.000000\n",
       "11       Sl  2014      9    40.689655    0.275862\n",
       "12       Sl  2014     10    70.928571    3.535714\n",
       "13       Sl  2014     11    75.238095    0.571429\n",
       "14       Sl  2014     12    41.000000    2.200000"
      ]
     },
     "execution_count": 297,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ = df.groupby([df.country, df.date.dt.year, df.date.dt.month]).mean()\n",
    "df_.index.names = ['country', 'year', 'month']\n",
    "df_.reset_index().sort_values(ascending=[True, True, True], by=['country', 'year', 'month'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Date</th>\n",
       "      <th>Variable</th>\n",
       "      <th>National</th>\n",
       "      <th>Bomi County</th>\n",
       "      <th>Bong County</th>\n",
       "      <th>Gbarpolu County</th>\n",
       "      <th>Grand Bassa</th>\n",
       "      <th>Grand Cape Mount</th>\n",
       "      <th>Grand Gedeh</th>\n",
       "      <th>Grand Kru</th>\n",
       "      <th>Lofa County</th>\n",
       "      <th>Margibi County</th>\n",
       "      <th>Maryland County</th>\n",
       "      <th>Montserrado County</th>\n",
       "      <th>Nimba County</th>\n",
       "      <th>River Gee County</th>\n",
       "      <th>RiverCess County</th>\n",
       "      <th>Sinoe County</th>\n",
       "      <th>Unnamed: 18</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Specimens collected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Specimens pending for testing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total specimens tested</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Newly reported deaths</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total death/s in confirmed cases</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total death/s in probable cases</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total death/s in suspected cases</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total death/s in confirmed, \\r\\n probable, sus...</td>\n",
       "      <td>3161.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>384.0</td>\n",
       "      <td>549.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1651.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Case Fatality Rate (CFR) - \\r\\n Confirmed &amp; Pr...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Newly reported contacts</td>\n",
       "      <td>263.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>181.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total contacts listed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Currently under follow-up</td>\n",
       "      <td>6101.0</td>\n",
       "      <td>142.0</td>\n",
       "      <td>218.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>410.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>1729.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>3297.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Contacts seen</td>\n",
       "      <td>5941.0</td>\n",
       "      <td>142.0</td>\n",
       "      <td>218.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>402.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>1613.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>3291.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Contacts who completed 21 day \\r\\n follow-up</td>\n",
       "      <td>859.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>499.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>264.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Contacts lost to follow-up</td>\n",
       "      <td>93.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>New admissions</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total no. currently in Treatment \\r\\n Units</td>\n",
       "      <td>149.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total discharges</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Cumulative admission/isolation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Newly Reported Cases in HCW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Cumulative cases among HCW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Newly Reported deaths in HCW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Cumulative deaths among HCW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>New Case/s (Suspected)</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>New Case/s (Probable)</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>New case/s (confirmed)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total suspected cases</td>\n",
       "      <td>3059.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>391.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>171.0</td>\n",
       "      <td>415.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1735.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total probable cases</td>\n",
       "      <td>1793.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>149.0</td>\n",
       "      <td>478.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>784.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total confirmed cases</td>\n",
       "      <td>2823.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>325.0</td>\n",
       "      <td>360.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1595.0</td>\n",
       "      <td>114.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total Number of Confirmed Cases \\r\\n of Sierra...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Total Number of Confirmed Cases \\r\\n of Guinea...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>12/1/2014</td>\n",
       "      <td>Cumulative confirmed, probable and suspected c...</td>\n",
       "      <td>7675.0</td>\n",
       "      <td>292.0</td>\n",
       "      <td>558.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>146.0</td>\n",
       "      <td>140.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>645.0</td>\n",
       "      <td>1253.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>4114.0</td>\n",
       "      <td>322.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Date                                           Variable  National  \\\n",
       "0   12/1/2014                                Specimens collected       NaN   \n",
       "1   12/1/2014                      Specimens pending for testing       NaN   \n",
       "2   12/1/2014                             Total specimens tested       NaN   \n",
       "3   12/1/2014                              Newly reported deaths       NaN   \n",
       "4   12/1/2014                   Total death/s in confirmed cases       NaN   \n",
       "5   12/1/2014                    Total death/s in probable cases       NaN   \n",
       "6   12/1/2014                   Total death/s in suspected cases       NaN   \n",
       "7   12/1/2014  Total death/s in confirmed, \\r\\n probable, sus...    3161.0   \n",
       "8   12/1/2014  Case Fatality Rate (CFR) - \\r\\n Confirmed & Pr...       NaN   \n",
       "9   12/1/2014                            Newly reported contacts     263.0   \n",
       "10  12/1/2014                              Total contacts listed       NaN   \n",
       "11  12/1/2014                          Currently under follow-up    6101.0   \n",
       "12  12/1/2014                                      Contacts seen    5941.0   \n",
       "13  12/1/2014       Contacts who completed 21 day \\r\\n follow-up     859.0   \n",
       "14  12/1/2014                         Contacts lost to follow-up      93.0   \n",
       "15  12/1/2014                                     New admissions      12.0   \n",
       "16  12/1/2014        Total no. currently in Treatment \\r\\n Units     149.0   \n",
       "17  12/1/2014                                   Total discharges       6.0   \n",
       "18  12/1/2014                     Cumulative admission/isolation       NaN   \n",
       "19  12/1/2014                        Newly Reported Cases in HCW       NaN   \n",
       "20  12/1/2014                         Cumulative cases among HCW       NaN   \n",
       "21  12/1/2014                       Newly Reported deaths in HCW       NaN   \n",
       "22  12/1/2014                        Cumulative deaths among HCW       NaN   \n",
       "23  12/1/2014                             New Case/s (Suspected)      25.0   \n",
       "24  12/1/2014                              New Case/s (Probable)       9.0   \n",
       "25  12/1/2014                             New case/s (confirmed)       1.0   \n",
       "26  12/1/2014                              Total suspected cases    3059.0   \n",
       "27  12/1/2014                               Total probable cases    1793.0   \n",
       "28  12/1/2014                              Total confirmed cases    2823.0   \n",
       "29  12/1/2014  Total Number of Confirmed Cases \\r\\n of Sierra...       NaN   \n",
       "30  12/1/2014  Total Number of Confirmed Cases \\r\\n of Guinea...       NaN   \n",
       "31  12/1/2014  Cumulative confirmed, probable and suspected c...    7675.0   \n",
       "\n",
       "    Bomi County  Bong County  Gbarpolu County  Grand Bassa  Grand Cape Mount  \\\n",
       "0           NaN          NaN              NaN          NaN               NaN   \n",
       "1           NaN          NaN              NaN          NaN               NaN   \n",
       "2           NaN          NaN              NaN          NaN               NaN   \n",
       "3           NaN          NaN              NaN          NaN               NaN   \n",
       "4           NaN          NaN              NaN          NaN               NaN   \n",
       "5           NaN          NaN              NaN          NaN               NaN   \n",
       "6           NaN          NaN              NaN          NaN               NaN   \n",
       "7         158.0        124.0             10.0         63.0              74.0   \n",
       "8           NaN          NaN              NaN          NaN               NaN   \n",
       "9           0.0         16.0              0.0         11.0              19.0   \n",
       "10          NaN          NaN              NaN          NaN               NaN   \n",
       "11        142.0        218.0              8.0         50.0             410.0   \n",
       "12        142.0        218.0              8.0         55.0             402.0   \n",
       "13          0.0          9.0              5.0         23.0               7.0   \n",
       "14          0.0          0.0              0.0          2.0               1.0   \n",
       "15          0.0          2.0              0.0          0.0               0.0   \n",
       "16          5.0         11.0              0.0          0.0               0.0   \n",
       "17          2.0          2.0              0.0          0.0               0.0   \n",
       "18          NaN          NaN              NaN          NaN               NaN   \n",
       "19          NaN          NaN              NaN          NaN               NaN   \n",
       "20          NaN          NaN              NaN          NaN               NaN   \n",
       "21          NaN          NaN              NaN          NaN               NaN   \n",
       "22          NaN          NaN              NaN          NaN               NaN   \n",
       "23          0.0          3.0              0.0          2.0               NaN   \n",
       "24          0.0          0.0              0.0          1.0               NaN   \n",
       "25          0.0          0.0              0.0          0.0               0.0   \n",
       "26         84.0        391.0             25.0         43.0              45.0   \n",
       "27         73.0         32.0              3.0         65.0              38.0   \n",
       "28        135.0        135.0             10.0         38.0              57.0   \n",
       "29          NaN          NaN              NaN          NaN               NaN   \n",
       "30          NaN          NaN              NaN          NaN               NaN   \n",
       "31        292.0        558.0             38.0        146.0             140.0   \n",
       "\n",
       "    Grand Gedeh  Grand Kru  Lofa County  Margibi County  Maryland County  \\\n",
       "0           NaN        NaN          NaN             NaN              NaN   \n",
       "1           NaN        NaN          NaN             NaN              NaN   \n",
       "2           NaN        NaN          NaN             NaN              NaN   \n",
       "3           NaN        NaN          NaN             NaN              NaN   \n",
       "4           NaN        NaN          NaN             NaN              NaN   \n",
       "5           NaN        NaN          NaN             NaN              NaN   \n",
       "6           NaN        NaN          NaN             NaN              NaN   \n",
       "7           5.0       26.0        384.0           549.0             17.0   \n",
       "8           NaN        NaN          NaN             NaN              NaN   \n",
       "9           0.0        0.0          0.0            34.0              0.0   \n",
       "10          NaN        NaN          NaN             NaN              NaN   \n",
       "11          1.0       72.0         66.0          1729.0             27.0   \n",
       "12          1.0       60.0         66.0          1613.0             27.0   \n",
       "13          0.0        0.0          0.0           499.0             46.0   \n",
       "14          0.0       12.0          0.0            17.0              0.0   \n",
       "15          0.0        0.0          0.0             1.0              0.0   \n",
       "16          0.0        0.0          0.0            20.0              0.0   \n",
       "17          0.0        0.0          0.0             0.0              0.0   \n",
       "18          NaN        NaN          NaN             NaN              NaN   \n",
       "19          NaN        NaN          NaN             NaN              NaN   \n",
       "20          NaN        NaN          NaN             NaN              NaN   \n",
       "21          NaN        NaN          NaN             NaN              NaN   \n",
       "22          NaN        NaN          NaN             NaN              NaN   \n",
       "23          2.0        0.0          0.0             0.0              5.0   \n",
       "24          0.0        0.0          0.0             0.0              1.0   \n",
       "25          NaN        0.0          0.0             0.0              0.0   \n",
       "26          8.0       14.0        171.0           415.0             15.0   \n",
       "27          0.0       17.0        149.0           478.0              2.0   \n",
       "28          2.0        4.0        325.0           360.0              4.0   \n",
       "29          NaN        NaN          NaN             NaN              NaN   \n",
       "30          NaN        NaN          NaN             NaN              NaN   \n",
       "31         10.0       35.0        645.0          1253.0             21.0   \n",
       "\n",
       "    Montserrado County  Nimba County  River Gee County  RiverCess County  \\\n",
       "0                  NaN           NaN               NaN               NaN   \n",
       "1                  NaN           NaN               NaN               NaN   \n",
       "2                  NaN           NaN               NaN               NaN   \n",
       "3                  NaN           NaN               NaN               NaN   \n",
       "4                  NaN           NaN               NaN               NaN   \n",
       "5                  NaN           NaN               NaN               NaN   \n",
       "6                  NaN           NaN               NaN               NaN   \n",
       "7               1651.0          53.0               8.0              25.0   \n",
       "8                  NaN           NaN               NaN               NaN   \n",
       "9                181.0           0.0               0.0               0.0   \n",
       "10                 NaN           NaN               NaN               NaN   \n",
       "11              3297.0           0.0               0.0               0.0   \n",
       "12              3291.0           0.0               0.0               0.0   \n",
       "13               264.0           0.0               0.0               6.0   \n",
       "14                 0.0           0.0               0.0               0.0   \n",
       "15                 9.0           0.0               0.0               0.0   \n",
       "16               105.0           8.0               0.0               0.0   \n",
       "17                 2.0           0.0               0.0               0.0   \n",
       "18                 NaN           NaN               NaN               NaN   \n",
       "19                 NaN           NaN               NaN               NaN   \n",
       "20                 NaN           NaN               NaN               NaN   \n",
       "21                 NaN           NaN               NaN               NaN   \n",
       "22                 NaN           NaN               NaN               NaN   \n",
       "23                 0.0          12.0               0.0               0.0   \n",
       "24                 0.0           7.0               0.0               0.0   \n",
       "25                 0.0           1.0               0.0               0.0   \n",
       "26              1735.0          80.0               7.0               9.0   \n",
       "27               784.0         128.0               5.0              16.0   \n",
       "28              1595.0         114.0               7.0              19.0   \n",
       "29                 NaN           NaN               NaN               NaN   \n",
       "30                 NaN           NaN               NaN               NaN   \n",
       "31              4114.0         322.0              19.0              44.0   \n",
       "\n",
       "    Sinoe County  Unnamed: 18  \n",
       "0            NaN          NaN  \n",
       "1            NaN          NaN  \n",
       "2            NaN          NaN  \n",
       "3            NaN          NaN  \n",
       "4            NaN          NaN  \n",
       "5            NaN          NaN  \n",
       "6            NaN          NaN  \n",
       "7           14.0          NaN  \n",
       "8            NaN          NaN  \n",
       "9            2.0          NaN  \n",
       "10           NaN          NaN  \n",
       "11          81.0          NaN  \n",
       "12          58.0          NaN  \n",
       "13           0.0          NaN  \n",
       "14          23.0          NaN  \n",
       "15           0.0          NaN  \n",
       "16           0.0          NaN  \n",
       "17           0.0          NaN  \n",
       "18           NaN          NaN  \n",
       "19           NaN          NaN  \n",
       "20           NaN          NaN  \n",
       "21           NaN          NaN  \n",
       "22           NaN          NaN  \n",
       "23           0.0          1.0  \n",
       "24           0.0          0.0  \n",
       "25           0.0          0.0  \n",
       "26          17.0          NaN  \n",
       "27           3.0          NaN  \n",
       "28          18.0          NaN  \n",
       "29           NaN          NaN  \n",
       "30           NaN          NaN  \n",
       "31          38.0          NaN  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv('C:\\\\Users\\\\Dario\\\\Documents\\\\Universit\\\\ADA\\\\Tutorials\\\\02 - Intro to Pandas\\\\Data\\\\ebola\\\\liberia_data\\\\2014-12-01.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 2. RNA Sequences\n",
    "\n",
    "In the `DATA_FOLDER/microbiome` subdirectory, there are 9 spreadsheets of microbiome data that was acquired from high-throughput RNA sequencing procedures, along with a 10<sup>th</sup> file that describes the content of each. \n",
    "\n",
    "Use pandas to import the first 9 spreadsheets into a single `DataFrame`.\n",
    "Then, add the metadata information from the 10<sup>th</sup> spreadsheet as columns in the combined `DataFrame`.\n",
    "Make sure that the final `DataFrame` has a unique index and all the `NaN` values have been replaced by the tag `unknown`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Write your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 3. Class War in Titanic\n",
    "\n",
    "Use pandas to import the data file `Data/titanic.xls`. It contains data on all the passengers that travelled on the Titanic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from IPython.core.display import HTML\n",
    "HTML(filename=DATA_FOLDER+'/titanic.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each of the following questions state clearly your assumptions and discuss your findings:\n",
    "1. Describe the *type* and the *value range* of each attribute. Indicate and transform the attributes that can be `Categorical`. \n",
    "2. Plot histograms for the *travel class*, *embarkation port*, *sex* and *age* attributes. For the latter one, use *discrete decade intervals*. \n",
    "3. Calculate the proportion of passengers by *cabin floor*. Present your results in a *pie chart*.\n",
    "4. For each *travel class*, calculate the proportion of the passengers that survived. Present your results in *pie charts*.\n",
    "5. Calculate the proportion of the passengers that survived by *travel class* and *sex*. Present your results in *a single histogram*.\n",
    "6. Create 2 equally populated *age categories* and calculate survival proportions by *age category*, *travel class* and *sex*. Present your results in a `DataFrame` with unique index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Write your answer here"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
