{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Task-1.-Compiling-Ebola-Data\"><span class=\"toc-item-num\">Task 1.&nbsp;&nbsp;</span>Compiling Ebola Data</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-2.-RNA-Sequences\"><span class=\"toc-item-num\">Task 2.&nbsp;&nbsp;</span>RNA Sequences</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-3.-Class-War-in-Titanic\"><span class=\"toc-item-num\">Task 3.&nbsp;&nbsp;</span>Class War in Titanic</a></div></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob\n",
    "import os\n",
    "from dateutil import parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "DATA_FOLDER = '../../ADA2017-Tutorials/02 - Intro to Pandas/Data/' # Use the data folder provided in Tutorial 02 - Intro to Pandas."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 1. Compiling Ebola Data\n",
    "\n",
    "The `DATA_FOLDER/ebola` folder contains summarized reports of Ebola cases from three countries (Guinea, Liberia and Sierra Leone) during the recent outbreak of the disease in West Africa. For each country, there are daily reports that contain various information about the outbreak in several cities in each country.\n",
    "\n",
    "Use pandas to import these data files into a single `Dataframe`.\n",
    "Using this `DataFrame`, calculate for *each country*, the *daily average per month* of *new cases* and *deaths*.\n",
    "Make sure you handle all the different expressions for *new cases* and *deaths* that are used in the reports."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Explanation\n",
    "A quick analysis of the dataset reveals that there are some inconsistencies, and therefore the first step will require some cleaning and some transformation steps. We can identify the following problems:\n",
    "- The columns have different names among different countries. For example, the Guinea dataset contains the columns: \"Date,Description,Totals\", whereas Liberia contains \"Date,Variable,National\" (the columns have the same meaning).\n",
    "- The CSV files have a variable number of columns. Fortunately, this is not an issue in our case, since we are only interested in totals.\n",
    "- There are many missing values. These will be treated as 0.\n",
    "- Dates are given in different formats, such as '2014-08-04' or '6/16/2014'.\n",
    "- Some records contain newlines. This issue is automatically handled by Pandas.\n",
    "- Row descriptions are inconsistent and are provided under different names. For instance, in Guinea we can find the following variables:\n",
    "    - New cases of suspects\n",
    "    - New cases of probables\n",
    "    - New cases of confirmed\n",
    "    - Total new cases registered so far *(the sum of the previous ones)*\n",
    "    \n",
    "    Conversely, in Liberia we can observe:\n",
    "    - New Case/s (Suspected)\n",
    "    - New Case/s (Probable)\n",
    "    - New case/s (confirmed) *(notice the difference between \"Case\" and \"case\")*\n",
    "    \n",
    "    There are other examples of these irregularities. They will be matched together.\n",
    "    \n",
    "### Plan\n",
    "The task will be subdivided into three parts:\n",
    "1. Dataset cleaning. This subtask will produce a consistent, uniform dataset.\n",
    "2. Pivoting: \"new cases\" and \"new death\" will be added as new columns.\n",
    "3. Aggregation: it will return the required daily averages per month.\n",
    "\n",
    "### Remarks\n",
    "Since it was not specified, we made some assumptions: by \"new cases\" we mean the sum of suspect, probable, and confirmed cases. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Load the dataset\n",
    "country_paths = glob.glob(DATA_FOLDER + 'ebola/*/')\n",
    "country_names = [os.path.basename(os.path.normpath(x)).replace('_data', '').capitalize() for x in country_paths]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Guinea', 'Liberia', 'Sl']\n"
     ]
    }
   ],
   "source": [
    "# Country names are derived from the directory name (N.B.: Sl = Sierra Leone)\n",
    "print(country_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# As explained earlier, we need to map different aliases to a standard column name\n",
    "aliases = {\n",
    "    'variable': 'description',\n",
    "    'national': 'totals'\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>country</th>\n",
       "      <th>description</th>\n",
       "      <th>totals</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>new cases of suspects</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>new cases of probables</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>new cases of confirmed</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>total new cases registered so far</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>total cases of suspects</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        date country                        description  totals\n",
       "0 2014-08-04  Guinea              new cases of suspects     5.0\n",
       "1 2014-08-04  Guinea             new cases of probables     0.0\n",
       "2 2014-08-04  Guinea             new cases of confirmed     4.0\n",
       "3 2014-08-04  Guinea  total new cases registered so far     9.0\n",
       "4 2014-08-04  Guinea            total cases of suspects    11.0"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = []\n",
    "for i, path in enumerate(country_paths):\n",
    "    filenames = glob.glob(path + \"/*.csv\")\n",
    "    for f in filenames:\n",
    "        df = pd.read_csv(f)\n",
    "        df.columns = [c.lower() for c in df.columns] # Set all columns to lower-case\n",
    "        df.rename(columns=aliases, inplace=True)\n",
    "        df['country'] = country_names[i]\n",
    "        df['date'] = [parser.parse(x, tzinfos={'dayfirst': True}) for x in df['date']] # Parse date\n",
    "        df['totals'] = pd.to_numeric(df['totals'], errors='coerce')\n",
    "        df = df.fillna(0) # Fill empty values with zeros\n",
    "        dfs.append(df[['date', 'country', 'description', 'totals']]) # Extract only relevant columns\n",
    "        \n",
    "df = pd.concat(dfs)\n",
    "df['description'] = [x.lower() for x in df['description']]\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Now we extract the relevant variables and we map them to a standard name (new_cases and new_deaths)\n",
    "column_keywords = {\n",
    "    'new_cases': ['new cases of suspects', 'new cases of probables', 'new cases of confirmed', 'new case/s (suspected)',\n",
    "                 'new case/s (probable)', 'new case/s (confirmed)',  'new_suspected', 'new_probable', 'new_confirmed'],\n",
    "    'new_deaths': ['new deaths registered today', 'new deaths registered', 'newly reported deaths',\n",
    "                  'etc_new_deaths']\n",
    "}\n",
    "\n",
    "desc_map = []\n",
    "for k, v in column_keywords.items():\n",
    "    for alias in v:\n",
    "        desc_map.append((k, alias))\n",
    "desc_map = pd.DataFrame(desc_map, columns=['type', 'description'])\n",
    "df = df.merge(desc_map, on='description', how='inner')\n",
    "df = df.drop('description', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>country</th>\n",
       "      <th>totals</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>5.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2014-08-26</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>18.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2014-08-27</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>12.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2014-08-30</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>15.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014-08-31</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>9.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2014-09-02</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>11.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2014-09-04</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>13.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2014-09-07</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>5.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2014-09-08</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>5.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2014-09-09</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>9.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2014-09-11</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>9.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2014-09-14</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>0.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2014-09-16</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>4.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2014-09-17</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>4.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2014-09-19</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>3.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2014-09-21</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>1.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2014-09-22</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>5.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2014-09-23</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>4.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2014-09-24</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>8.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2014-09-26</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>2.0</td>\n",
       "      <td>new_cases</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         date country  totals       type\n",
       "0  2014-08-04  Guinea     5.0  new_cases\n",
       "1  2014-08-26  Guinea    18.0  new_cases\n",
       "2  2014-08-27  Guinea    12.0  new_cases\n",
       "3  2014-08-30  Guinea    15.0  new_cases\n",
       "4  2014-08-31  Guinea     9.0  new_cases\n",
       "5  2014-09-02  Guinea    11.0  new_cases\n",
       "6  2014-09-04  Guinea    13.0  new_cases\n",
       "7  2014-09-07  Guinea     5.0  new_cases\n",
       "8  2014-09-08  Guinea     5.0  new_cases\n",
       "9  2014-09-09  Guinea     9.0  new_cases\n",
       "10 2014-09-11  Guinea     9.0  new_cases\n",
       "11 2014-09-14  Guinea     0.0  new_cases\n",
       "12 2014-09-16  Guinea     4.0  new_cases\n",
       "13 2014-09-17  Guinea     4.0  new_cases\n",
       "14 2014-09-19  Guinea     3.0  new_cases\n",
       "15 2014-09-21  Guinea     1.0  new_cases\n",
       "16 2014-09-22  Guinea     5.0  new_cases\n",
       "17 2014-09-23  Guinea     4.0  new_cases\n",
       "18 2014-09-24  Guinea     8.0  new_cases\n",
       "19 2014-09-26  Guinea     2.0  new_cases"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# This is much better\n",
    "df.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# The dataframe is pivoted so as to transform new_cases and new_deaths types to columns\n",
    "# Furthermore, equivalent types are aggregated with a sum (e.g. suspected + probable + confirmed)\n",
    "pivoted = df.pivot_table(index=['date', 'country'], columns='type', values='totals', aggfunc='sum')\n",
    "pivoted.columns.name = None\n",
    "df = pivoted.reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>country</th>\n",
       "      <th>new_cases</th>\n",
       "      <th>new_deaths</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2014-06-16</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2014-06-17</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2014-06-22</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>10.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2014-06-24</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2014-06-25</td>\n",
       "      <td>Liberia</td>\n",
       "      <td>7.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        date  country  new_cases  new_deaths\n",
       "0 2014-06-16  Liberia        4.0         2.0\n",
       "1 2014-06-17  Liberia        2.0         0.0\n",
       "2 2014-06-22  Liberia       10.0         4.0\n",
       "3 2014-06-24  Liberia        6.0         4.0\n",
       "4 2014-06-25  Liberia        7.0         3.0"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Some records from Liberia\n",
    "df[df.country == 'Liberia'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>country</th>\n",
       "      <th>year</th>\n",
       "      <th>month</th>\n",
       "      <th>new_cases</th>\n",
       "      <th>new_deaths</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>25.800000</td>\n",
       "      <td>3.400000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>2014</td>\n",
       "      <td>9</td>\n",
       "      <td>19.625000</td>\n",
       "      <td>3.562500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>2014</td>\n",
       "      <td>10</td>\n",
       "      <td>34.000000</td>\n",
       "      <td>15.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>6</td>\n",
       "      <td>5.714286</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>7</td>\n",
       "      <td>8.545455</td>\n",
       "      <td>4.272727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>37.222222</td>\n",
       "      <td>23.222222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>9</td>\n",
       "      <td>63.833333</td>\n",
       "      <td>36.041667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>10</td>\n",
       "      <td>45.560000</td>\n",
       "      <td>28.040000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>11</td>\n",
       "      <td>26.466667</td>\n",
       "      <td>13.466667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>2014</td>\n",
       "      <td>12</td>\n",
       "      <td>5178.555556</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Sl</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>25.150000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Sl</td>\n",
       "      <td>2014</td>\n",
       "      <td>9</td>\n",
       "      <td>40.689655</td>\n",
       "      <td>0.275862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Sl</td>\n",
       "      <td>2014</td>\n",
       "      <td>10</td>\n",
       "      <td>70.928571</td>\n",
       "      <td>3.535714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Sl</td>\n",
       "      <td>2014</td>\n",
       "      <td>11</td>\n",
       "      <td>75.238095</td>\n",
       "      <td>0.571429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Sl</td>\n",
       "      <td>2014</td>\n",
       "      <td>12</td>\n",
       "      <td>41.000000</td>\n",
       "      <td>2.200000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    country  year  month    new_cases  new_deaths\n",
       "0    Guinea  2014      8    25.800000    3.400000\n",
       "1    Guinea  2014      9    19.625000    3.562500\n",
       "2    Guinea  2014     10    34.000000   15.000000\n",
       "3   Liberia  2014      6     5.714286    2.000000\n",
       "4   Liberia  2014      7     8.545455    4.272727\n",
       "5   Liberia  2014      8    37.222222   23.222222\n",
       "6   Liberia  2014      9    63.833333   36.041667\n",
       "7   Liberia  2014     10    45.560000   28.040000\n",
       "8   Liberia  2014     11    26.466667   13.466667\n",
       "9   Liberia  2014     12  5178.555556    0.000000\n",
       "10       Sl  2014      8    25.150000    0.000000\n",
       "11       Sl  2014      9    40.689655    0.275862\n",
       "12       Sl  2014     10    70.928571    3.535714\n",
       "13       Sl  2014     11    75.238095    0.571429\n",
       "14       Sl  2014     12    41.000000    2.200000"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Group by country, year, month, and aggregate with the mean value. This produces the daily average per country/year/month.\n",
    "df_ = df.groupby([df.country, df.date.dt.year, df.date.dt.month]).mean()\n",
    "df_.index.names = ['country', 'year', 'month']\n",
    "df_.reset_index().sort_values(ascending=[True, True, True], by=['country', 'year', 'month'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Final observations\n",
    "The number of new cases in Liberia, in December 2014, seems too high. However, a manual inspection revealed that this figure is reflected in the original dataset."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 2. RNA Sequences\n",
    "\n",
    "In the `DATA_FOLDER/microbiome` subdirectory, there are 9 spreadsheets of microbiome data that was acquired from high-throughput RNA sequencing procedures, along with a 10<sup>th</sup> file that describes the content of each. \n",
    "\n",
    "Use pandas to import the first 9 spreadsheets into a single `DataFrame`.\n",
    "Then, add the metadata information from the 10<sup>th</sup> spreadsheet as columns in the combined `DataFrame`.\n",
    "Make sure that the final `DataFrame` has a unique index and all the `NaN` values have been replaced by the tag `unknown`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAXON</th>\n",
       "      <th>COUNT</th>\n",
       "      <th>BARCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>7</td>\n",
       "      <td>MID1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Thermopro...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>7</td>\n",
       "      <td>MID1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               TAXON  COUNT BARCODE\n",
       "0  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...      7    MID1\n",
       "1  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...      2    MID1\n",
       "2  Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...      3    MID1\n",
       "3  Archaea \"Crenarchaeota\" Thermoprotei Thermopro...      3    MID1\n",
       "4  Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...      7    MID1"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path = DATA_FOLDER + \"microbiome\"\n",
    "metadata = glob.glob(path + \"/metadata.xls\")\n",
    "filenames = glob.glob(path + \"/MID*.xls\")\n",
    "\n",
    "rnaData = []\n",
    "for i, file in enumerate(filenames):\n",
    "    currData = pd.read_excel(file, header=None, names=[\"TAXON\", \"COUNT\"])\n",
    "    currData[\"BARCODE\"] = \"MID\"+str(i+1) # Add a column value to do the join\n",
    "    rnaData.append(currData)\n",
    "\n",
    "result = pd.concat(rnaData)\n",
    "result.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAXON</th>\n",
       "      <th>COUNT</th>\n",
       "      <th>BARCODE</th>\n",
       "      <th>GROUP</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>7</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Thermopro...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>7</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               TAXON  COUNT BARCODE  \\\n",
       "0  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...      7    MID1   \n",
       "1  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...      2    MID1   \n",
       "2  Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...      3    MID1   \n",
       "3  Archaea \"Crenarchaeota\" Thermoprotei Thermopro...      3    MID1   \n",
       "4  Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...      7    MID1   \n",
       "\n",
       "                GROUP SAMPLE  \n",
       "0  EXTRACTION CONTROL    NaN  \n",
       "1  EXTRACTION CONTROL    NaN  \n",
       "2  EXTRACTION CONTROL    NaN  \n",
       "3  EXTRACTION CONTROL    NaN  \n",
       "4  EXTRACTION CONTROL    NaN  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Perform an inner join on BARCODE, so as to match the metadata with the actual content\n",
    "metadata = pd.read_excel(metadata[0])\n",
    "result = pd.merge(result, metadata, how=\"inner\", on=\"BARCODE\")\n",
    "result.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TAXON      0\n",
      "COUNT      0\n",
      "BARCODE    0\n",
      "GROUP      0\n",
      "SAMPLE     0\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# Replace missing values (NaNs) with \"unknown\"\n",
    "result = result.fillna(\"unknown\")\n",
    "# Make sure that there are no NaNs\n",
    "print(pd.isnull(result).sum()) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>COUNT</th>\n",
       "      <th>GROUP</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TAXON</th>\n",
       "      <th>BARCODE</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Firmicutes\" \"Clostridia\" Clostridiales Veillonellaceae Schwartzia</th>\n",
       "      <th>MID3</th>\n",
       "      <td>1</td>\n",
       "      <td>Control 1</td>\n",
       "      <td>tissue</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Actinobacteria\" Actinobacteria Actinomycetales Promicromonosporaceae Xylanibacterium</th>\n",
       "      <th>MID1</th>\n",
       "      <td>2</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Proteobacteria\" Gammaproteobacteria \"Enterobacteriales\" Enterobacteriaceae Samsonia</th>\n",
       "      <th>MID5</th>\n",
       "      <td>237</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>tissue</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Verrucomicrobia\" Opitutae Puniceicoccales Puniceicoccaceae Cerasicoccus</th>\n",
       "      <th>MID7</th>\n",
       "      <td>39</td>\n",
       "      <td>Control 1</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Bacteroidetes\" \"Bacteroidia\" \"Bacteroidales\" \"Rikenellaceae\" Rikenella</th>\n",
       "      <th>MID7</th>\n",
       "      <td>1</td>\n",
       "      <td>Control 1</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Bacteroidetes\" Flavobacteria \"Flavobacteriales\" Flavobacteriaceae Nonlabens</th>\n",
       "      <th>MID4</th>\n",
       "      <td>1</td>\n",
       "      <td>NEC 2</td>\n",
       "      <td>tissue</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Actinobacteria\" Actinobacteria Actinomycetales Microbacteriaceae Microbacterium</th>\n",
       "      <th>MID2</th>\n",
       "      <td>1</td>\n",
       "      <td>NEC 1</td>\n",
       "      <td>tissue</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Firmicutes\" \"Clostridia\" Clostridiales Veillonellaceae Propionispira</th>\n",
       "      <th>MID8</th>\n",
       "      <td>6</td>\n",
       "      <td>NEC 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Bacteroidetes\" Flavobacteria \"Flavobacteriales\" Flavobacteriaceae Zeaxanthinibacter</th>\n",
       "      <th>MID2</th>\n",
       "      <td>1</td>\n",
       "      <td>NEC 1</td>\n",
       "      <td>tissue</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacteria \"Firmicutes\" \"Bacilli\" Bacillales Bacillaceae Thalassobacillus</th>\n",
       "      <th>MID8</th>\n",
       "      <td>1</td>\n",
       "      <td>NEC 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                            COUNT  \\\n",
       "TAXON                                              BARCODE          \n",
       "Bacteria \"Firmicutes\" \"Clostridia\" Clostridiale... MID3         1   \n",
       "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         2   \n",
       "Bacteria \"Proteobacteria\" Gammaproteobacteria \"... MID5       237   \n",
       "Bacteria \"Verrucomicrobia\" Opitutae Puniceicocc... MID7        39   \n",
       "Bacteria \"Bacteroidetes\" \"Bacteroidia\" \"Bactero... MID7         1   \n",
       "Bacteria \"Bacteroidetes\" Flavobacteria \"Flavoba... MID4         1   \n",
       "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID2         1   \n",
       "Bacteria \"Firmicutes\" \"Clostridia\" Clostridiale... MID8         6   \n",
       "Bacteria \"Bacteroidetes\" Flavobacteria \"Flavoba... MID2         1   \n",
       "Bacteria \"Firmicutes\" \"Bacilli\" Bacillales Baci... MID8         1   \n",
       "\n",
       "                                                                         GROUP  \\\n",
       "TAXON                                              BARCODE                       \n",
       "Bacteria \"Firmicutes\" \"Clostridia\" Clostridiale... MID3              Control 1   \n",
       "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
       "Bacteria \"Proteobacteria\" Gammaproteobacteria \"... MID5              Control 2   \n",
       "Bacteria \"Verrucomicrobia\" Opitutae Puniceicocc... MID7              Control 1   \n",
       "Bacteria \"Bacteroidetes\" \"Bacteroidia\" \"Bactero... MID7              Control 1   \n",
       "Bacteria \"Bacteroidetes\" Flavobacteria \"Flavoba... MID4                  NEC 2   \n",
       "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID2                  NEC 1   \n",
       "Bacteria \"Firmicutes\" \"Clostridia\" Clostridiale... MID8                  NEC 2   \n",
       "Bacteria \"Bacteroidetes\" Flavobacteria \"Flavoba... MID2                  NEC 1   \n",
       "Bacteria \"Firmicutes\" \"Bacilli\" Bacillales Baci... MID8                  NEC 2   \n",
       "\n",
       "                                                             SAMPLE  \n",
       "TAXON                                              BARCODE           \n",
       "Bacteria \"Firmicutes\" \"Clostridia\" Clostridiale... MID3      tissue  \n",
       "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     unknown  \n",
       "Bacteria \"Proteobacteria\" Gammaproteobacteria \"... MID5      tissue  \n",
       "Bacteria \"Verrucomicrobia\" Opitutae Puniceicocc... MID7       stool  \n",
       "Bacteria \"Bacteroidetes\" \"Bacteroidia\" \"Bactero... MID7       stool  \n",
       "Bacteria \"Bacteroidetes\" Flavobacteria \"Flavoba... MID4      tissue  \n",
       "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID2      tissue  \n",
       "Bacteria \"Firmicutes\" \"Clostridia\" Clostridiale... MID8       stool  \n",
       "Bacteria \"Bacteroidetes\" Flavobacteria \"Flavoba... MID2      tissue  \n",
       "Bacteria \"Firmicutes\" \"Bacilli\" Bacillales Baci... MID8       stool  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# seT an index that is unique (the combination of TAXON and BARCODE is suitable)\n",
    "result.set_index([\"TAXON\",\"BARCODE\"], inplace=True)\n",
    "print(result.index.is_unique) # Check uniqueness\n",
    "\n",
    "# Show a sample of the transformed dataset\n",
    "result.sample(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 3. Class War in Titanic\n",
    "\n",
    "Use pandas to import the data file `Data/titanic.xls`. It contains data on all the passengers that travelled on the Titanic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.core.display import HTML\n",
    "HTML(filename=DATA_FOLDER+'/titanic.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each of the following questions state clearly your assumptions and discuss your findings:\n",
    "1. Describe the *type* and the *value range* of each attribute. Indicate and transform the attributes that can be `Categorical`. \n",
    "2. Plot histograms for the *travel class*, *embarkation port*, *sex* and *age* attributes. For the latter one, use *discrete decade intervals*. \n",
    "3. Calculate the proportion of passengers by *cabin floor*. Present your results in a *pie chart*.\n",
    "4. For each *travel class*, calculate the proportion of the passengers that survived. Present your results in *pie charts*.\n",
    "5. Calculate the proportion of the passengers that survived by *travel class* and *sex*. Present your results in *a single histogram*.\n",
    "6. Create 2 equally populated *age categories* and calculate survival proportions by *age category*, *travel class* and *sex*. Present your results in a `DataFrame` with unique index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Write your answer here"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
